### PR TITLE
Refactor lodash imports

### PR DESCRIPTION
**Pull request by truongnguyen-eh**
### Description
 The current import causes the full lodash build to be included in the bundle of the apps that are using this library.

### Feature
- Use selective lodash imports
- Use rollup in replacement of webpack to generate both cjs and esm
- Use jest in placement of mocha

### Test
image

### CODE REVIEW

File: rollup.config.js [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+    'lodash/isEqual',
+    'lodash/toString',
+    'lodash/trim',
+    'lodash/findIndex',
+    'lodash/concat',
+    'lodash/join',
+    'lodash/curry',
+    'lodash/cloneDeep',
+    'lodash/set',
+  ],
```
Comment (sangquangnguyen) [inline rollup.config.js]: could we set 
```
external:['lodash']
```
e?

Comment (truongnguyen-eh) [reply rollup.config.js]: @sangquangnguyen We can use Regex `external: /lodash\/.*/,`, I just finished the update, thanks for you suggestion anh 💯

**Review by sangquangnguyen — APPROVED** [review]

File: rollup.config.js (hunk 1 above)
Comment (sangquangnguyen) [reply rollup.config.js]: just my curiosity, why can't we just set `external:['lodash']` @truongnguyen-eh  e?

Comment (truongnguyen-eh) [reply rollup.config.js]: @sangquangnguyen Looks like we need to pass the exact imports which are used inside `src/index.js` into `external` config as [mentioned](https://rollupjs.org/configuration-options/#external) in the document. I attempted to apply `external:['lodash']` but received the warning from termial a
image

**Review by phthhieu — APPROVED** [review]

File: rollup.config.js (hunk 1 above)
Comment (sangquangnguyen) [reply rollup.config.js]: Thanks @truongnguyen-eh 🚀

**Pull request merged**